urlc98c3b97d2_Trimax_venta_tgz-pJ8-com_gesoftware_venta_structures_map_MapTestJ8/src/com/gesoftware/venta/utility/ApplicationUtility.java

```diff
@@ -0,0 +1,125 @@
+package com.gesoftware.venta.utility;
+
+import com.gesoftware.venta.logging.LoggingUtility;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+
+/**
+ * ApplicationUtility class definition
+ **/
+public final class ApplicationUtility {
+    /** The lock_file. */
+    private File m_LockFile = null;
+
+    /** The lock. */
+    private FileLock m_Lock = null;
+
+    /** The lock_channel. */
+    private FileChannel m_LockChannel = null;
+
+    /** The lock_stream. */
+    private FileOutputStream m_LockStream = null;
+
+    /* *
+     * METHOD: Application utility class constructor
+     *  PARAM: [IN] key - application name
+     * AUTHOR: Eliseev Dmitry
+     * */
+    private ApplicationUtility(final String key) throws Exception {
+        String tmp_dir = System.getProperty("java.io.tmpdir");
+        if (!tmp_dir.endsWith(System.getProperty("file.separator")))
+            tmp_dir += System.getProperty("file.separator");
+
+        /* Trying to lock file */
+        try {
+            m_LockFile = new File(tmp_dir + HashUtility.generateHash(key) + ".app_lock");
+        } catch (Exception e) {
+            LoggingUtility.core("AppLock.AppLock() file fail: " + e.getMessage());
+        }
+
+        /* MD5 acquire fail */
+        if (m_LockFile == null)
+            m_LockFile = new File(tmp_dir + key + ".app_lock");
+
+        m_LockStream = new FileOutputStream(m_LockFile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource that might leak to a class field, causing a resource leak that slows down or crashes your system. For example, when FileChannel.tryLock throws an exception, the *ApplicationUtility* object isn’t constructed so you can’t close this resource afterward.

**Fix**
Ensure that if an exception is thrown, the *m\_LockStream* resource is closed before the constructor is exited.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).